### DISCUSSION TITLE
Error In POST_PRODUCT_PRICING_DATA in BRAZIL market place

### DISCUSSION THREAD

**Discussion by kapildel** [abuzuhri/Amazon-SP-API-CSharp#457] [Q&A]
Hi, I am Creating feed for POST_PRODUCT_PRICING_DATA using below code
`Dim CFS As New ConstructFeedService(_MerchantId, "1.00")
                Dim ProductLst As New List(Of PriceMessage)
                Dim irowCount As Integer = 0
                For irowCount = 0 To ds.Tables("Info").Rows.Count - 1
                    Dim PL As New PriceMessage
                    Dim SPI As New StandardPrice
                    PL.SKU = "100000X"
                    SPI.currency = "USD"
                    SPI.Value = "99.99"
                    PL.StandardPrice = SPI
                    ProductLst.Add(PL)
                Next
                CFS.AddPriceMessage(ProductLst)
                sXmlSent = CFS.GetXML()
                Dim MPI As New Utils.MarketplaceIds
                MPI.Add(_MarketPlaceId)
                sAZFeedId = amzcon.Feed.SubmitFeed(sXmlSent, FeedType.POST_PRODUCT_PRICING_DATA, MPI, Nothing, ContentType.XML)`

the xml post on amazon  in case Of US locale
 
""<?xml version="1.0" encoding="utf-16"?>
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Header>
    <DocumentVersion>2.1.0</DocumentVersion>
    <MerchantIdentifier>XXXXXXXX</MerchantIdentifier>
  </Header>
  <MessageType>Price</MessageType>
  <Message>
    <MessageID>1</MessageID>
    <OperationType>Update</OperationType>
    <Price>
      <SKU>100000X</SKU>
      <StandardPrice currency="USD">309.11</StandardPrice>
    </Price>
  </Message>
</AmazonEnvelope>""

and this data processed sucessfully.

But in case of BR xml generated by code is

`<?xml version="1.0" encoding="utf-16"?>
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Header>
    <DocumentVersion>2.1.0</DocumentVersion>
    <MerchantIdentifier>XXXXXXXX</MerchantIdentifier>
  </Header>
  <MessageType>Price</MessageType>
  <Message>
    <MessageID>1</MessageID>
    <OperationType>Update</OperationType>
    <Price>
      <SKU>100000X</SKU>
      <StandardPrice currency="BRL">712,03</StandardPrice>
    </Price>
  </Message>
</AmazonEnvelope>`

and this message Id return Error in result
**XML Parsing Error at Line 59, Column cvc-datatype-valid.1.2.1: '712,03' is not a valid value for 'decimal'.: .**

The only difference i can see in xml dot and comma used in price value to seperate decimals.

when i post with Dot in price value in case of BR its working fine.

Should i replace the comma with dot in price value of is there any other solution?

**Comment by KoesterRobert** [comment]
StandardPrice:value is a string
Try 
string price = (712.03M).ToString("0.00", System.Globalization.CultureInfo.CreateSpecificCulture("en-EN"))
to format it properly
